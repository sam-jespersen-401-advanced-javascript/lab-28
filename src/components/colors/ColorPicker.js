import React from 'react';
import PropTypes from 'prop-types';

const ColorPicker = ({ colors, selectColor }) => {
  const colorButtons = colors.map(color => (
    <button key={color} onClick={() => selectColor(color)}>{color}</button>
  ));
  return (
    <>
      {colorButtons}
    </>
  );
};

ColorPicker.propTypes = {
  colors: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectColor: PropTypes.func.isRequired
};

export default ColorPicker;
