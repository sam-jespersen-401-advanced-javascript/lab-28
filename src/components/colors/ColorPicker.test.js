import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker component', () => {
  let selectColor = jest.fn();
  it('renders ColorPicker', () => {
    const wrapper = shallow(<ColorPicker colors={['red']} selectColor={selectColor} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls selectColor function onclick', () => {
    const wrapper = shallow(<ColorPicker colors={['red', 'green', 'purple']} selectColor={selectColor} />);
    wrapper.findWhere(el => el.key() === 'green').simulate('click');
    expect(selectColor).toHaveBeenCalledWith('green');
  });

});
