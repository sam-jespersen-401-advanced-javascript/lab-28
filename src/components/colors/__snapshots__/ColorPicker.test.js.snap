// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders ColorPicker 1`] = `
<Fragment>
  <button
    key="red"
    onClick={[Function]}
  >
    red
  </button>
</Fragment>
`;
